src/ui/src/components/shared/SharedImgWithFallback.vue

```diff
@@ -1,31 +1,87 @@
 <template>
-	<img :src="src" />
+	<WdsSkeletonLoader v-if="isLoading" class="skeleton" :style="loaderStyle" />
+	<img v-else :src="src" />
 </template>
 
 <script lang="ts" setup>
+import {
+	type CSSProperties,
+	computed,
+	type PropType,
+	ref,
+	toRef,
+	watch,
+} from "vue";
 import { useAssetContentType } from "@/composables/useAssetContentType";
-import { PropType, ref, toRef, watch } from "vue";
+import WdsSkeletonLoader from "@/wds/WdsSkeletonLoader.vue";
 
 const props = defineProps({
 	urls: { type: Array as PropType<string[]>, required: true },
+	loaderMaxWidthPx: { type: Number, default: undefined },
+	loaderMaxHeightPx: { type: Number, default: undefined },
+});
+
+const loaderStyle = computed(() => {
+	const style: CSSProperties = {};
+	if (props.loaderMaxWidthPx !== undefined) {
+		style.maxWidth = `${props.loaderMaxWidthPx}px`;
+		style.width = `${props.loaderMaxWidthPx}px`;
+	}
+	if (props.loaderMaxHeightPx !== undefined) {
+		style.maxHeight = `${props.loaderMaxHeightPx}px`;
+		style.height = `${props.loaderMaxHeightPx}px`;
+	}
+	return style;
 });
 
 const src = ref("");
+const isLoading = ref(true);
+const previousUrls = ref<string[]>([]);
+let currentGeneration = 0;
 
 const { fetchAssetContentType } = useAssetContentType();
 
+// Helper function to check if arrays are deeply equal
+function areUrlsEqual(a: string[], b: string[]): boolean {
+	if (a.length !== b.length) return false;
+	return a.every((url, index) => url === b[index]);
+}
+
 watch(
 	toRef(props, "urls"),
 	async (urls) => {
+		// Skip if URLs haven't changed
+		if (areUrlsEqual(urls, previousUrls.value)) {
+			return;
+		}
```

Comment:
I'm not sure if you need to use a special ref for `previousUrls`, you can use the previous value from the watcher callback

```suggestion
	async (urls, previousUrls) => {
		// Skip if URLs haven't changed
		if (areUrlsEqual(urls, previousUrls)) {
			return;
		}
```